service/worker/replicator/replicationTask.go

```diff
@@ -433,43 +297,21 @@ func (t *historyMetadataReplicationTask) Execute() error {
 			t.version,
 			t.nextEventID,
 			t.version)
-	} else {
-		return t.historyRereplicator.SendMultiWorkflowHistory(
-			t.queueID.NamespaceID, t.queueID.WorkflowID,
-			t.queueID.RunID, t.firstEventID,
-			t.queueID.RunID, t.nextEventID,
-		)
 	}
+
+	return serviceerror.NewInternal("historyMetadataReplicationTask encounter error which cannot be handled")
 }
 
 func (t *historyMetadataReplicationTask) HandleErr(
 	err error,
 ) error {
-	retryErr, ok := err.(*serviceerrors.RetryTask)
-	if !ok || retryErr.RunId == "" {
+	retryErr, ok := err.(*serviceerrors.RetryTaskV2)
+	if !ok {
 		return err
 	}
 
-	t.metricsClient.IncCounter(metrics.HistoryRereplicationByHistoryReplicationScope, metrics.ClientRequests)
-	stopwatch := t.metricsClient.StartTimer(metrics.HistoryRereplicationByHistoryReplicationScope, metrics.ClientLatency)
-	defer stopwatch.Stop()
-
-	// this is the retry error
-	beginRunID := retryErr.RunId
-	beginEventID := retryErr.NextEventId
-	endRunID := t.queueID.RunID
-	endEventID := t.taskID
-	resendErr := t.historyRereplicator.SendMultiWorkflowHistory(
-		t.queueID.NamespaceID, t.queueID.WorkflowID,
-		beginRunID, beginEventID, endRunID, endEventID,
-	)
-	if resendErr != nil {
-		t.logger.Error("error resend history", tag.Error(resendErr))
-		// should return the replication error, not the resending error
-		return err
-	}
-	// should try again
-	return t.Execute()
```

Comment:
should use the ndc logic for resend

Reply:
I'll file a bug to add this logic later.  Even the existing code is missing resend logic for ndc.

Reply:
Filed #700